src/NuGet.Core/NuGet.Logging/CollectorLogger.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace NuGet.Logging
+{
+    public class CollectorLogger : ICollectorLogger
```

Comment:
Ah, I see, that may happen. This will get hit many times in a large restore such as Roslyn or CoreFX. It could be worth the perf improvement to just limit it to errors for now. The name can stay the same.